python/python/lancedb/query.py

```diff
@@ -895,40 +926,69 @@ class LanceHybridQueryBuilder(LanceQueryBuilder):
     def __init__(
         self,
         table: "Table",
-        query: str,
-        vector_column: str,
+        query: str = None,
+        vector_column: str = None,
         fts_columns: Union[str, List[str]] = [],
     ):
         super().__init__(table)
-        vector_query, fts_query = self._validate_query(query)
-        self._fts_query = LanceFtsQueryBuilder(
-            table, fts_query, fts_columns=fts_columns
-        )
-        vector_query = self._query_to_vector(table, vector_query, vector_column)
-        self._vector_query = LanceVectorQueryBuilder(table, vector_query, vector_column)
+        self._query = query
+        self._vector_column = vector_column
+        self._fts_columns = fts_columns
         self._norm = "score"
         self._reranker = LinearCombinationReranker(weight=0.7, fill=1.0)
+        self._nprobes = None
+        self._refine_factor = None
 
-    def _validate_query(self, query):
-        # Temp hack to support vectorized queries for hybrid search
-        if isinstance(query, str):
-            return query, query
-        elif isinstance(query, tuple):
-            if len(query) != 2:
-                raise ValueError(
-                    "The query must be a tuple of (vector_query, fts_query)."
-                )
-            if not isinstance(query[0], (list, np.ndarray, pa.Array, pa.ChunkedArray)):
-                raise ValueError(f"The vector query must be one of {VEC}.")
-            if not isinstance(query[1], str):
-                raise ValueError("The fts query must be a string.")
-            return query[0], query[1]
-        else:
+    def _validate_query(self, query, vector=None, text=None):
+        if query is not None and vector is not None:
```

Comment:
if we don't allow the users to pass both query & vector, we should ban text as well?

Reply:
Actually I wasn't sure about this based on [this comment](https://github.com/lancedb/lancedb/pull/1013#pullrequestreview-1913154943) (see the last usage). But your suggestion sounds better as it'll be less confusing for the user. Updated it!